src/software/amazon/ion/impl/lite/IonSequenceLite.java

```diff
@@ -398,4 +400,330 @@ void writeBodyTo(IonWriter writer, SymbolTableProvider symbolTableProvider)
+            size++;
+        }
+
+        public boolean addAll(final Collection<? extends IonValue> c) {
+            for (IonValue ionValue : c){
+                add(ionValue);
+            }
+
+            return true;
+        }
+
+        public boolean addAll(final int index, final Collection<? extends IonValue> c) {
+            int i = index;
+
+            for (IonValue ionValue : c){
+                add(i, ionValue);
+                i++;
+            }
+
+            return true;
+        }
+
+        public boolean retainAll(final Collection<?> c) {
+            if (size < 1) {
+                return false;
+            }
+
+            final IdentityHashMap<Object, Object> toKeep = new IdentityHashMap<Object, Object>();
+
+            for (Object o : c) {
+                if (indexOf(o) > 0) {
+                    toKeep.put(o, o);
+                }
+            }
+
+            if (toKeep.size() == size) {
+                return false;
+            }
+
+            for (int i = 0; i < size; i++) {
+                if(!toKeep.containsKey(get(i))) {
+                    remove(i);
+                }
+            }
+
+            return true;
+        }
+
+        public void clear() {
+            checkForParentModification();
+
+            final List<IonValue> toRemove = new ArrayList<IonValue>(size);
+            for(int i = 0; i < size; i++) {
+                toRemove.add(get(i));
+            }
+
+            parent.removeAll(toRemove);
+
+            size = 0;
+            this.structuralModificationCount = IonSequenceLite.this.structuralModificationCount;
+        }
+
+        public IonValue remove(final int index) {
+            rangeCheck(index);
+            checkForParentModification();
+
+            final IonValue removed = parent.remove(toParentIndex(index));
+
+            this.structuralModificationCount = IonSequenceLite.this.structuralModificationCount;
+            size--;
+
+            return removed;
+        }
+
+        public boolean remove(final Object o) {
+            int index = indexOf(o);
+            if(index < 0) {
+                return false;
+            }
+
+            remove(index);
+            return true;
+        }
+
+        public boolean removeAll(final Collection<?> c) {
+            boolean changed = false;
+            for (Object o : c) {
+                if (remove(o)) {
+                    changed = true;
+                }
+            }
+
+            return changed;
+        }
+
+        public int indexOf(final Object o) {
+            checkForParentModification();
+
+            final int parentIndex = parent.indexOf(o);
+            final int index = fromParentIndex(parentIndex);
+
+            // not found
+            if(parentIndex < 0 || index < 0 || index >= size) {
+                return -1;
+            }
+
+            return index;
+        }
+
+        public int lastIndexOf(final Object o) {
+            return indexOf(o);
```

Comment:
Se my answer to your comments on the test for `indexOf`. In short for `IonSequence` those two are identical since `IonSequence`s don't support duplicates